### ISSUE TITLE
Manipulating in w+space

### ISSUE THREAD

**Issue by yilmazkorkmz** [genforce/higan#3]
Thank you for great work. I came here from interfacegan repo.

When i try w+,w or z space manipulation on generated images the results seem so good. I am using my trained stylegan2 network. However, when i tried the system with real images by projecting them to latent space (to w+space), It seems the boundary search  does not work as well as before although i am trying to search boundary for same feature with same scorer function.  In boundary search accuracy stuck around 50 for all layers. I tried several things:

1) I trained boundary search on 14,000 real data (projecting 14000 real image) instead of choosing 2000 best samples.
2) I tried using the same boundary which i found by using generated images with real images but it seems creating artifacts. 

I am using stylegan2 official inversion network for projecting real images with small differences. (converted it to find different codes for different layers and added a pixelwise loss) almost same as https://github.com/Puzer/stylegan-encoder.

Thanks in advance.

I added the console output for layer4 and 5 it is almost same for all layers:
[2020-05-18 04:19:06,596][INFO] ==== Layer 04 ====
[2020-05-18 04:19:06,604][INFO] Spliting training and validation sets:
[2020-05-18 04:19:06,604][INFO]   Num of data after filtering: 14028.
[2020-05-18 04:19:06,612][INFO]   Training set (9818): 4909 positive, 4909 negative.
[2020-05-18 04:19:06,612][INFO]   Validation set (4210): 2105 positive, 2105 negative.
[2020-05-18 04:19:06,612][INFO] Training boundary.
[2020-05-18 06:18:55,934][INFO] Finish training.
[2020-05-18 06:19:13,465][INFO] Accuracy for validation set:
[2020-05-18 06:19:13,466][INFO]   Positive: 1041 / 2105 = 0.494537
[2020-05-18 06:19:13,466][INFO]   Negative: 1078 / 2105 = 0.512114
[2020-05-18 06:19:13,466][INFO]   All: 2119 / 4210 = 0.503325
[2020-05-18 06:19:13,469][INFO] ==== Layer 05 ====
[2020-05-18 06:19:13,478][INFO] Spliting training and validation sets:
[2020-05-18 06:19:13,478][INFO]   Num of data after filtering: 14028.
[2020-05-18 06:19:13,486][INFO]   Training set (9818): 4909 positive, 4909 negative.
[2020-05-18 06:19:13,486][INFO]   Validation set (4210): 2105 positive, 2105 negative.
[2020-05-18 06:19:13,486][INFO] Training boundary.
[2020-05-18 08:22:26,411][INFO] Finish training.
[2020-05-18 08:22:43,720][INFO] Accuracy for validation set:
[2020-05-18 08:22:43,720][INFO]   Positive: 1014 / 2105 = 0.481710
[2020-05-18 08:22:43,721][INFO]   Negative: 1039 / 2105 = 0.493587
[2020-05-18 08:22:43,721][INFO]   All: 2053 / 4210 = 0.487648

**Comment by ShenYujun** [comment]
The stylegan2 official inversion seems to also use the layer-wise noises for pixel reconstruction. This will significantly affect the semantic information contained in the inverted code. Please see [this repo](https://github.com/genforce/idinvert) for more details.

**Comment by yilmazkorkmz** [comment]
Actually, i turned off noise optimization, network only optimizes latents so there should be no difference with puzer's encoder. Do you have any idea about cause other from inversion network?

I try that repo as well, thanks for sharing.

**Comment by ShenYujun** [comment]
1. You can first check whether the predicted score is correct.
2. You can choose to only use the data with the highest confidence (i,e., the one with the highest scores and the lowest scores).
3. This thing may be caused by the discrepancy between the real domain and the synthesis domain.
